### DISCUSSION TITLE
Shutting down TODO

### DISCUSSION THREAD

**Discussion by JasonEtco** [JasonEtco/todo#293] [Announcements]
Hello! I come bearing bad news - I've decided to shut down TODO. I'm doing that for two reasons:

* Up until now, I've been hosting the application on GitHub's infrastructure, and that's no longer tennable for security reasons.
* I haven't had the bandwidth or motivation to properly support this project, and its both unfair and counterproductive to leave it in an ambiguously abandoned state.

So, soon the application will go into maintenance mode, and will be unavailable for a few days - with the plan being to shut it down completely shortly thereafter. At that point, I'll be archiving this repository - the code will remain open source, so if you're interesting in hosting it yourself, you can do so (though don't expect support going forward).

### What are your alternatives?

Well, TODO was created before [GitHub Actions](https://github.com/features/actions). If I were to rewrite it today, it would be an Action instead of an external application. I'd start there. There are [some alternatives out there in the wild already](https://github.com/marketplace?type=actions&query=todo+).

### Can someone else take over this project?

You're welcome to fork the codebase and do whatever you want with it. But, as far as existing installations of the app goes, I will not be transferring ownership, and will be deleting the GitHub App after the application is shut down.

If you have concerns, feel free to comment below, but know that this decision is not going to change. I appreciate the support over the duration of this project though, and I'm sorry that (as all things must), it has to end ❤️ 

---

* **Friday, Dec 10**: I've put the Heroku app into maintenance mode.

**Comment by thislooksfun** [comment]
You mention security concerns, can you elaborate on that at all? I want to make sure my projects don't suffer from the same problem(s).

**Reply by JasonEtco** [reply]
Sure - it's not security concerns of this project, but I'm not able to continue hosting this application on GitHub's internal infrastructure for security reasons. Namely, anything on GitHub's infra requires security review(s) and consistent maintenance, neither of which I can support for this project.

To be clear: nothing about the code or anything has security problems (that I know of).

**Reply by thislooksfun** [reply]
Ah, I see. I misunderstood what you meant, but that makes perfect sense.

**Comment by rugk** [comment]
Okay the advantage in contrast to GitHub Actions-based solutions is that it applied to all repositories… that was quite convenient.

**Reply by JasonEtco** [reply]
I agree, that is a pitfall of Actions. GitHub recently released [Reusable Workflow](https://docs.github.com/en/actions/learn-github-actions/reusing-workflows), which is a (imperfect IMO) solution to that.

**Comment by bilalshaikh42** [comment]
Thanks for your work on this and for maintaining it as long as you did!
